rfc006-release-planning-mgt.md

```diff
@@ -0,0 +1,88 @@
+# RFC006 - Release Planning & Management Process
+
+## Proposed by
+
+Aaron Peters (@acpkendo)
+
+## Current status
+
+- [x] Draft
+- [x] Under discussion (until 2021-09-15)
+- [ ] Final comment and voting (until YYYY-MM-DD) {{Add date after selecting this status.}}
+- [ ] Accepted
+- [ ] Rejected
+- [ ] Implemented
+- [ ] Deferred
+- [ ] Withdrawn
+
+
+## Proposal overview
+
+This proposal outlines a process to plan for and manage releases for The Good Docs Project. Specifically, it will detail:
+
+- The process through which the project defines and agrees on a Release;
+- The versioning scheme the project will use to track and communicate progress across Releases, and;
+- The tooling the project will use to manage progress towards Releases.

```

Comment:
💭 ❔ __thoughts/questions:_ Thought I'd kick us off 💨 

- I have questions around many areas I feel should be included so like @barbaricyawps did in her #10 she noted what was going to be out of scope which helped curtail some of those questions I immediately had.
- ❔ Which projects will need to be included in the Release Coordination? How do those Projects/Working-groups communicate that they need to be added to the Release process for any upcoming releases.  Or can a working-group release on their own cadence independently?  Are there rules around how all this works/communicated?
- ➡️ Versioning Scheme -- I didn't see any concrete notes on the process here in the proposal so not sure what is in scope/out-of-scope.  But think this should potentially be included here.  Happy to provide language/thoughts here as this is probably in my wheel-house.
- ➡️ Tooling, other than the mention of Zenhub as our project-management.  A simple version of this might be bundling/tagging releases on github.  And that our default tooling is github's packaging.
- 📣 How do we communicate releases to external/consumers and the _marketing-team_.  If we are setting a regular cadence of releases we should communicate that.
- 💭 Btw, I'm a fan of [keep-a-changelog](https://keepachangelog.com/en/1.0.0/) as one way to manage some release-notes even if there are deliverables.

Reply:
Hi @mgan59, I made some changes to facilitate discussion, a fair amount of which I feel will still be required.

First question: I think all WGs/sub-projects should be eligible to be included in a release, but I think it's a mistake to say "all of them must" or "releases consist of precisely X." An Agile approach to this would allow the individual teams to decide what they think is the most valuable items (within their own purview) are, as well as what they can confidently commit to in order to meet a given release (most likely time-bounded, to speak to the point raised by @barbaricyawps below). But making requirements such as "all releases contain updates to the templates" or "every WG must contribute something to a release" will inevitably put us in a position where e.g. a WG is short on help, can't get a quality contribution ready in time, and then what? IMO each release should be a negotiated agreement between the PSC (i.e. "the business") on what they want to reach the project's high-level goals, and the WGs (i.e. the Agile teams) on what they feel is most valuable and can deliver.

Second point: Versioning... I've volunteered to draft a separate RFC for this, which in my mind makes sense as the number attached to deliverables isn't dependent on the process used to plan for those deliverables, and vice versa. Once I'm ready to start on this draft I will certainly work with you on its details.

Third point: Tooling... I added a section to describe the toolset and its roles more clear, although it probably needs a little more yet.

Fourth point: Marketing... This should arguably be a different RFC, as it's a different function performed by a different person (Product Marketing Manager) in many organizations. I also feel the organization would benefit from some sort of marketing/PR chair/team to promote the project across a variety of channels. Stated more simply, I feel like this is something the marketing team (once it exists) should drive.